pandas/core/nanops.py

```diff
@@ -107,21 +107,9 @@ def f(values, axis=None, skipna=True, **kwds):
                     if k not in kwds:
                         kwds[k] = v
             try:
-                if values.size == 0:
-
-                    # we either return np.nan or pd.NaT
-                    if is_numeric_dtype(values):
-                        values = values.astype('float64')
-                    fill_value = na_value_for_dtype(values.dtype)
-
-                    if values.ndim == 1:
-                        return fill_value
-                    else:
-                        result_shape = (values.shape[:axis] +
-                                        values.shape[axis + 1:])
-                        result = np.empty(result_shape, dtype=values.dtype)
-                        result.fill(fill_value)
-                        return result
+                if values.size == 0 and kwds.get('min_count') is None:
+                    # We are empty, returning NA for our type
```

Comment:
isn’t min_count always passed?

Reply:
No, just when the caller is `sum` or `prod`. Methods like `mean` don't pass it.

Reply:
I find this very error prone. The internal functions can *always* pass ``min_count`` I think, so it will be not-None. or at the very least comment on this.

Reply:
The issue here passing through the correct arguments to  the nanop. `pandas.core.nanops.sum` takes `min_count`, while `mean_doesn't`. The function calling `f` knows that, but `f` doesn't. So I would either

1. refactor `f` to *not* do this special casing, and have each nanop properly handle size-0 arrays
2. add an unused `min_count` to each nanop, and always pass it through
3. have `f` add `min_count` to `kwds` when it's not None.

I think 1 is the best option, but will take a bit of time. 3 doesn't strike me as any better than the current method.

Reply:
I think you could just pass it thru in ``make_stat_function``? e.g. .you always just pass

```
        # not sum/prod (I know you have a separate function for this)
        min_count=-1
        return self._reduce(f, name, axis=axis, skipna=skipna,
                            numeric_only=numeric_only, min_count=min_count)
```

Reply:
That'd be option 2. I'd have to add a `min_count` keyword to each of our methods in `pandas.core.nanops`, which I don't think we want.